environment/src/main/java/com/sequenceiq/environment/network/service/SubnetIdProvider.java

```diff
@@ -24,12 +29,41 @@
     private CloudPlatformConnectors cloudPlatformConnectors;
 
     public String provide(NetworkDto network, Tunnel tunnel, CloudPlatform cloudPlatform) {
-        LOGGER.debug("Choosing subnet, network: {},  platform: {}, tunnel: {}", network, cloudPlatform, tunnel);
         if (network == null || network.getSubnetIds() == null || network.getSubnetIds().isEmpty() || network.getCbSubnets() == null
                 || network.getCbSubnets().isEmpty()) {
             LOGGER.debug("Check failed, returning null");
             return null;
         }
+        return selectSubnet(network, tunnel, cloudPlatform, network.getCbSubnetValues(), null);
+    }
+
+    public String provideEndpointGateway(NetworkDto network, CloudPlatform cloudPlatform, String baseSubnetId) {
+        if (baseSubnetId == null || network == null ||
+                ((network.getEndpointGatewaySubnetMetas() == null || network.getEndpointGatewaySubnetMetas().isEmpty())
+                && (network.getCbSubnets() == null || network.getCbSubnets().isEmpty()))) {
+            LOGGER.debug("Check failed, returning null");
```

Comment:
Can we explain why the check failed, it would result in an easier root cause analyses from the logs??

Reply:
I'll update this in PR https://github.com/hortonworks/cloudbreak/pull/9675/

Reply:
Actually I checked and the changes in this class have all been removed from my PR. The logic to select the endpoint gateway subnet is instead being done in `EnvironmentBaseNetworkConverter`. Because the endpoint gateway subnet must be in the same AZ as the stack subnet, we need to wait until after that subnet selection is final before we can select the endpoint gateway subnet. So this logic was happening too early and was removed.